src/lib/routing-table.js

```diff
@@ -25,7 +25,10 @@ class RoutingTable {
     const routes = this.destinations.get(destination)
     if (!routes) return
     routes.delete(nextHop)
-    if (routes.size === 0) this.destinations.delete(destination)
+    if (routes.size === 0) {
+      this.destinations.delete(destination)
+      return -1
```

Comment:
`-1` isn't a very intuitive return value here.. maybe a `Boolean` of whether or not a destination was removed (and a function comment to be explicit).